doc/developer/platform/ux.md

```diff
@@ -1,20 +1,902 @@
+app-specific password for each external service that will connect to Materialize
+Platform via the PostgreSQL protocol.
+
+Rationale: the PostgreSQL protocol does not have separate fields for an API
+client ID and API secret key, so both values need to be packed into the password
+field. Naively packing the UUIDs is unwieldy, however:
+
+```
+893677c2-a74c-4ecb-804d-3b7fa0c61fdc:68104da2-3417-442e-a614-f078dac470be
+```
+
+But since UUIDs are just 16 bytes each, we can pack them into a much shorter
+base64-encoded string:
+
+```py
+>>> base64.urlsafe_b64encode(client_id.bytes + secret_key.bytes).decode().rstrip('=')
+iTZ3wqdMTsuATTt_oMYf3GgQTaI0F0QuphTweNrEcL4
+```
+
+When encoded in this form, we call the value an app-specific password rather
+than an API key.
+
+In the web UI, we will need to build a custom interface for managing personal
+API tokens that shows the app-specific password form alongside the client ID
+and secret.
+
+### Billing
+
+Materialize Platform will bill monthly based on usage. Accounts will be billed
+in terms of compute units, storage units, and network transfer units. A strawman
+proposal: compute units are virtual machine-hours, storage units are
+gigabyte-months, and network transfer units are gigabytes. Only outbound
+network traffic is billed.
+
+We expect metering the compute usage of clusters, the storage usage of sources
+and tables, and outbound network traffic generated by sinks and query responses,
+to be straightforward. What is not clear is how to bill for the overhead of
+other system components.
+
+### Region
+
+When an account is created, it does not have any regions. The user's first step
+after registering for a new account is to choose one region to activate.
+
+Additional regions can be enabled via a "+ Enable new region" button in the
+region selector or somesuch in the web UI, or via the [global API](#global-api).
+
+To prevent disaster, regions cannot be deactivated. Users who wish to deactivate
+a region can contact support.
+
+The initially supported regions will be the same regions as supported today
+by Materialize Cloud:
+
+  * aws:us-east-1
+  * aws:eu-west-1
+
+### Cluster
+
+Clusters are created within a region via a new `CREATE CLUSTER` SQL statement:
+
+```sql
+CREATE CLUSTER <name> SIZE {xs|s|m|l|xl} [, REPLICATION FACTOR <n>];
+```
+
+Creating a cluster allocates persistent compute resources (e.g., EC2 instances)
+and begins billing the account for those resources. The SQL statement returns
+successfully as soon as the request is received.
+
+Specifying a replication factor greater than one will run multiple identical
+replicas of the cluster for redundancy. Any requests to the cluster will be
+transparently routed to any live replica, so the cluster can tolerate failures
+of `n - 1` replicas before it becomes unavailable.
+
+A cluster may be horizontally scaled across multiple machines. The scaling is
+deliberately hidden from users behind the opaque notion of "size".
```

Comment:
I love that we're hiding users from worrying about scaling across multiple machines, but I'm curious what the bounds are for how much a cluster can dynamically scale? For example, if a user creates a cluster of size "small", and quickly approaches an OOM situation within the bounds of "small", do we automatically upgrade the cluster to medium, do we just keep scaling (regardless of how big it gets), or is this manual on the part of the user to create a bigger cluster?

Reply:
Oh I completely overlooked that! Scaling will be a manual process, but a (hopefully) zero-downtime one. In the far future I think we definitely want autoscaling, but not something we're presently ready to build.

I added this text to the cluster section:

> The size and replication factor of an existing cluster can be changed via the new
> `ALTER CLUSTER` SQL statement:
> 
> ```sql
> ALTER CLUSTER <name> [SIZE {xs|s|m|l|xl}] [, REPLICATION FACTOR <n>];
> ```
> 
> Changing either of these properties is a zero-downtime operation that happens
> asynchronously. As with creating a cluster, users can monitor the status of an
> `ALTER CLUSTER` operation via the `SHOW CLUSTERS` statement.
> 
> In the (far) future, we plan to build "autoscaling", in which users would
> specify a minimum and maximum size for their cluster, and Materialize Platform
> would automatically shrink or grow the cluster's size within those limits in
> response to observed resource utilization.

Reply:
Perfect, thanks!